web-api-db-ctrl/src/elastic.rs

```diff
@@ -78,16 +78,41 @@ pub(crate) async fn migrate_tenant_index(
     tenant_id: &TenantId,
     embedding_size: usize,
 ) -> Result<(), Error> {
-    if !does_tenant_index_exist(elastic, tenant_id).await? {
+    if let Some(existing_mapping) = get_opt_tenant_mapping(elastic, tenant_id).await? {
+        let base_mapping = mapping_with_embedding_size(&MAPPING, embedding_size)?;
+        check_mapping_compatibility(&existing_mapping, &base_mapping)?;
+    } else {
         error!(
             {%tenant_id},
             "index for tenant doesn't exist, creating a new index"
         );
         create_tenant_index(elastic, tenant_id, embedding_size).await?;
     }
-    //FIXME code to check if the index is a super-set of the expected index
-    //FIXME code for allowing updates to the ES schema, at least if
-    //      incremental application of the change is possible
+    Ok(())
+}
+
+fn check_mapping_compatibility(
+    existing_mapping: &Value,
+    base_mapping: &Value,
+) -> Result<(), Error> {
+    let existing_embeddings = &existing_mapping["mappings"]["properties"]["embedding"];
+    let expected_embeddings = &base_mapping["mappings"]["properties"]["embedding"];
```

Comment:
:art: we probably should introduce constants for these reused indexing strings to better guard against typos.